File: frontend/app/models/artist.js
import DS from 'ember-data';

export default DS.Model.extend({

  backgroundStyle: DS.attr(),
  bio: DS.attr(),
  facebookUrl: DS.attr(),
  headerImage: DS.attr(),
  keywords: DS.attr(),
  listImageUrl: DS.attr(),
  name: DS.attr(),
  publish: DS.attr('boolean'),
  albums: DS.hasMany('album'),
  slug: DS.attr(),
  spotifyUrl: DS.attr(),
  twitterUrl: DS.attr(),
  videos: DS.attr(),
  youtubeUrl: DS.attr(),

});

File: backend/lib/api/artists/artist.ex
defmodule FSM.Artists.Artist do
  use Ecto.Schema
  import Ecto.Changeset
  alias FSM.Artists.Artist
  alias FSM.Releases.Album
  alias FSM.Users.User

  schema "artists" do
    field :name, :string
    field :keywords, :string
    field :slug, :string
    field :list_image_url, :string
    field :background_style, :string
    field :header_image, :string
    field :bio, :string
    field :videos, :string
    field :facebook_url, :string
    field :twitter_url, :string
    field :youtube_url, :string
    field :spotify_url, :string
    field :publish, :boolean

    has_many :albums, Album
    many_to_many :users, User, join_through: "artists_users"
  end

  def changeset(%Artist{} = artist, attrs) do
    artist
    |> cast(attrs, [:name])
    |> validate_required([:name])
  end
end
